crates/rust-client/src/rpc/errors.rs

```diff
@@ -66,32 +43,19 @@ impl From<RpcConversionError> for RpcError {
 // RPC CONVERSION ERROR
 // ================================================================================================
 
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub enum RpcConversionError {
+    #[error("value is not in the range 0..modulus")]
     NotAValidFelt,
-    NoteTypeError(NoteError),
+    #[error("invalid note type value")]
+    NoteTypeError(#[source] NoteError),
+    #[error("field `{field_name}` required to be filled in protobuf representation of {entity}")]
```

Comment:
nit: Let's change this to something like:

```suggestion
    #[error("field `{field_name}` expected to be present in protobuf representation of {entity}")]
```